features/steps/app_steps.rb

```diff
@@ -1,25 +1,82 @@
+When('I run {string}') do |scenario_name|
+  execute_command :run_scenario, scenario_name
+end
+
+When("I run {string} and relaunch the crashed app") do |event_type|
+  steps %(
+    Given I run \"#{event_type}\"
+    And I relaunch the app after a crash
+  )
+end
+
+When("I run the configured scenario and relaunch the crashed app") do
+  platform = Maze::Helper.get_current_platform
+  case platform
+  when 'ios'
+    run_and_relaunch
+  when 'macos'
+    $scenario_mode = $last_scenario[:scenario_mode]
+    execute_command($last_scenario[:action], $last_scenario[:scenario_name])
+  else
+    raise "Unknown platform: #{platform}"
+  end
+end
+
+def run_and_relaunch
+  steps %(
+    Given I click the element "run_scenario"
+    And the app is not running
+    Then I relaunch the app
+  )
+end
+
+When('I clear all persistent data') do
+  platform = Maze::Helper.get_current_platform
+  case platform
+  when 'ios'
+    steps %(
+      When I click the element "clear_persistent_data"
+    )
+  when 'macos'
+    $reset_data = true
+  else
+    raise "Unknown platform: #{platform}"
+  end
+end
+
+When('I configure Bugsnag for {string}') do |scenario_name|
+  execute_command :start_bugsnag, scenario_name
+end
+
 When('I relaunch the app') do
   case Maze::Helper.get_current_platform
   when 'macos'
-    app = Maze.driver.capabilities['app']
-    system("killall -KILL #{app} > /dev/null && sleep 1")
-    Maze.driver.get(app)
+    # Pass
   else
     Maze.driver.launch_app
   end
 end
 
 When("I relaunch the app after a crash") do
   # Wait for the app to stop running before relaunching
-  step 'the app is not running'
   case Maze::Helper.get_current_platform
   when 'macos'
-    Maze.driver.get(Maze.driver.capabilities['app'])
+    # Allow any operations to complete before the next step
+    sleep(2)
```

Comment:
No - sleeping is not a reliable solution. We need to wait for the process to die.

Reply:
I think I gave a bad steer on this one - I'd said it was ok on the basis that overall the approach is still quicker than Appium-based and doesn't come with that baggage.  I didn't think about the reliability of the approach and that sometimes it may not be long enough.

Reply:
I did have a more concrete reason for this, but I've made other changes since then so I'll switch this back and re-test, and see if it's made a difference where we don't require the wait.

Reply:
But the tests should also (essentially) assert that the app did die in a crashing scenario.

Not dying can indicate a deadlock in the crash reporter.